phase1-coordinator/src/tests.rs

```diff
@@ -1300,6 +1301,386 @@ fn try_lock_blocked_test() -> anyhow::Result<()> {
+            _ => panic!("unexpected object type"),
+        };
+
+        // There is now no current contributor1
+        assert!(coordinator.current_contributor_info(&contributor1).unwrap().is_none());
+
+        // Check the coordinator state matches what is in storage.
+        assert!(
+            coordinator_state
+                .current_contributor_info(&contributor1)
+                .unwrap()
+                .is_none()
+        );
+
+        let dropped_participant1_info = coordinator_state
+            .dropped_contributors()
+            .into_iter()
+            .find(|contributor_info: &ParticipantInfo| contributor_info.participant() == contributor1)
+            .expect("expected participant1 to be present in the list of dropped contributors");
+
+        assert!(dropped_participant1_info.is_dropped());
+
+        let contributor2_info = coordinator
+            .current_contributor_info(&contributor2)
+            .unwrap()
+            .expect("expected contributor2 to have info now that it is a current contributor in round 1");
+
+        // Check the coordinator state matches what is in storage.
+        assert_eq!(
+            &contributor2_info,
+            coordinator_state
+                .current_contributor_info(&contributor2)
+                .unwrap()
+                .expect("expected contributor2 to have info now that it is a current contributor in round 1")
+        );
+
+        // There are now more assigned tasks than before the participant drop.
+        assert!(1 < contributor2_info.assigned_tasks().len());
+        assert!(contributor2_info.pending_tasks().is_empty());
+        // There are now less completed tasks than before the participant drop.
+        assert!(contributor2_n_tasks - 1 > contributor2_info.completed_tasks().len());
+        assert!(contributor2_info.disposing_tasks().is_empty());
+        // Some of the tasks that depended on participant1's
+        // contributions have been disposed.
+        assert!(!contributor2_info.disposed_tasks().is_empty());
+
+        let verifier_info = coordinator
+            .current_verifier_info(&verifier)
+            .unwrap()
+            .expect("expected verifier to have info now that it is a verifier in round 1");
+
+        // Check the coordinator state matches what is in storage.
+        assert_eq!(
+            &verifier_info,
+            coordinator_state
+                .current_verifier_info(&verifier)
+                .unwrap()
+                .expect("expected verifier to have info now that it is a current verifier in round 1")
+        );
+
+        assert!(verifier_info.pending_tasks().is_empty());
+        assert!(!verifier_info.completed_tasks().is_empty());
+        assert!(verifier_info.disposing_tasks().is_empty());
+        // Some of the tasks that depended on participant1's
+        // contributions have been disposed.
+        assert!(!verifier_info.disposed_tasks().is_empty());
```

Comment:
This part of the test is failing, perhaps this is an incorrect assumption?

Reply:
@raychu86 @howardwu would we expect that if a participant is dropped, the verifier will also move some of its completed tasks (which depend on that contributor) to the `disposed_tasks` list?

Reply:
@raychu86 @howardwu  sorry, bump